src/combinators/mod.rs

```diff
@@ -135,20 +123,20 @@ pub fn position<'a, F>(f: F) -> impl Fn(&'a mut [u8]) -> Result<(&'a mut [u8], &
 ///
 /// let mut buf = [0u8; 100];
 ///
-/// let len = {
-///   let (_, len) = pair(string("abcd"), string("efgh"))(&mut buf[..]).unwrap();
-///   len
-/// };
+/// {
+///   let buf = pair(string("abcd"), string("efgh"))(&mut buf[..]).unwrap();
+///   assert_eq!(buf.len(), 100 - 8);
+/// }
 ///
-/// assert_eq!(len, 8usize);
 /// assert_eq!(&buf[..8], &b"abcdefgh"[..]);
 /// ```
-pub fn pair<F, G, I>(first: F, second: G) -> impl SerializeFn<I>
+pub fn pair<F, G, I: Write>(first: F, second: G) -> impl SerializeFn<I>
 where F: SerializeFn<I>,
       G: SerializeFn<I> {
 
   move |out: I| {
-    first(out).chain(&second)
+    let out = first(out)?;
+    second(out)
```

Comment:
Why are you changing that?

Reply:
Changing it to `first(out).and_then(second)`, that's indeed clearer